### PR TITLE
add: Protected routes

### PR DESCRIPTION
**Pull request by devsurajrai**
added protected routes to the app using authentication. Please review the code and provide the feedback.

### CODE REVIEW

File: src/components/SideBar/side-bar.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+.sidebar-item{
+    display:flex;
+    align-items:center;
+}
+.sidebar-item span{
+    letter-spacing:3px;
+}
+.sidebar-btn-userinfo{
+   justify-content:space-between;
+   height:calc(100vh - 46rem);
```
Comment (palash23jain) [inline src/components/SideBar/side-bar.css:17]: good use of calc property

**Review by palash23jain — APPROVED** [review]
Code looks clean and organised.

File: src/components/RequiresAuth/RequiresAuth.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Navigate, useLocation } from "react-router-dom";
+import { useAuthContext } from "../../contexts/authContext";
+
+const RequiresAuth = ({ children }) => {
+  const { auth, test } = useAuthContext();
+  const location = useLocation();
+  console.log(location);
+  console.log(auth);
```
Comment (saifudinbadani) [inline src/components/RequiresAuth/RequiresAuth.jsx:8]: remove this part, if testing is done already.

**Review by saifudinbadani — APPROVED** [review]
Code looks fine to me.

**Pull request merged**